DataFormats/Luminosity/interface/PixelClusterCounts.h

```diff
@@ -14,40 +14,39 @@
 #include <iostream>
 #include <vector>
 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Luminosity/interface/LumiConstants.h"
+
+
 namespace reco {
 class PixelClusterCounts {
-static constexpr unsigned int nEmBX=3564;//Empty BX to fill with counts
+
     public:
-        PixelClusterCounts() : m_events(nEmBX){}
-        ////////////////////////////////////////////
-        void increment(int mD,int BXid,int count){
-            std::vector<int>::iterator it;
-            it = std::find(m_ModID.begin(), m_ModID.end(), mD);
-            size_t modIndex = it - m_ModID.begin();
-
-            if(it == m_ModID.end()){
-                std::vector<int> m_empBX(nEmBX, 0);
+        PixelClusterCounts() : m_events(LumiConstants::numBX){}
+
+        void increment(int mD,unsigned int bxID,int count){
+            size_t modIndex = std::distance(m_ModID.begin(), std::find(m_ModID.begin(), m_ModID.end(), mD));
+            if (modIndex == m_ModID.size()){
                 m_ModID.push_back(mD);
-                m_counts.insert(m_counts.begin(),m_empBX.begin(),m_empBX.end()); 
+                m_counts.resize(m_counts.size()+LumiConstants::numBX, 0);
             }
-            m_counts[nEmBX*modIndex+BXid] += count; 
+            m_counts.at(LumiConstants::numBX*modIndex+bxID-1) += count; 
         }
-        ////////////////////////////////////////////
-        void eventCounter(int BXid){
-            m_events[BXid]++;
+
+        void eventCounter(unsigned int  bxID){
+                m_events.at(bxID-1)++;
```

Comment:
The bxID does start at 1. Reran over file and printed out the bxID just to make sure.